tkg/client/update_credentials.go

```diff
@@ -197,3 +214,98 @@ func (c *TkgClient) updateVSphereCredentialsForCluster(clusterClient clusterclie
 
 	return nil
 }
+
+// UpdateAzureClusterCredentials update azure cluster credentials
+func (c *TkgClient) UpdateAzureClusterCredentials(clusterClient clusterclient.Client, options *UpdateCredentialsOptions) error {
+	if options.AzureUpdateClusterOptions.AzureTenantID == "" || options.AzureUpdateClusterOptions.AzureClientID == "" || options.AzureUpdateClusterOptions.AzureClientSecret == "" {
+		return errors.New("either tenantId, clientId or clientSecret should not be empty")
+	}
+
+	if err := c.updateAzureCredentialsForCluster(clusterClient, options); err != nil {
+		return err
+	}
+
+	if options.IsRegionalCluster {
+		if options.IsCascading {
+			log.Infof("Updating credentials for all workload clusters under management cluster %q", options.ClusterName)
+			clusters, err := clusterClient.ListClusters("")
```

Comment:
This api returns all clusters created by this mc, including itself?

Reply:
Yes. The clusterClient is created from RegionContexts which is for each MC.